diff --git a/app/include/data/custom_graph.hpp b/app/include/data/custom_graph.hpp
index 9483383..e8e87c7 100644
--- a/app/include/data/custom_graph.hpp
+++ b/app/include/data/custom_graph.hpp
@@ -41,5 +41,6 @@ protected:
     std::unordered_set<QtNodes::NodeId> m_funcOutNodes;
 
 private:
+    // show warning for invalid port connection (implicit typing failure)
     Q_INVOKABLE void showWarning(QtNodes::ConnectionId connectionId);
 };
diff --git a/app/include/ui/models/nodes.hpp b/app/include/ui/models/nodes.hpp
index d904b12..23fbd34 100644
--- a/app/include/ui/models/nodes.hpp
+++ b/app/include/ui/models/nodes.hpp
@@ -6,6 +6,7 @@
 
 using QtNodes::NodeData;
 using QtNodes::NodeDataType;
+using QtNodes::NodeId;
 
 class NamedNode : public NodeData
 {
@@ -28,6 +29,11 @@ protected:
     NodeDataType m_type;
 };
 
+struct TypeId {
+    NodeId nodeId;
+    PortIndex portIndex;
+};
+
 class DataNode : public NamedNode
 {
 public:
diff --git a/app/src/data/custom_graph.cpp b/app/src/data/custom_graph.cpp
index 34c2207..bca386b 100644
--- a/app/src/data/custom_graph.cpp
+++ b/app/src/data/custom_graph.cpp
@@ -4,6 +4,7 @@
 #include "ui/models/io_models.hpp"
 #include "ui/models/processor_models.hpp"
 #include <QMessageBox>
+#include <QAbstractButton>
 #include <QMetaObject>
 
 namespace {
@@ -63,7 +64,14 @@ void CustomGraph::showWarning(QtNodes::ConnectionId connectionId)
                                   .arg(connectionId.outPortIndex)
                                   .arg(connectionId.inNodeId)
                                   .arg(connectionId.inPortIndex));
-    msgBox.exec();
+    msgBox.setStandardButtons(QMessageBox::Ok | QMessageBox::Ignore);
+    msgBox.setDefaultButton(QMessageBox::Ok);
+    msgBox.button(QMessageBox::Ignore)->setText("Override");
+    int ret = msgBox.exec();
+    if (ret == QMessageBox::Ignore) {
+        qDebug() << "override" << connectionId.outNodeId;
+        // Handle the override logic here
+    }
 }
 
 void CustomGraph::warnInvalidConnection(QtNodes::ConnectionId const connectionId) const
diff --git a/app/src/ui/main_window.cpp b/app/src/ui/main_window.cpp
index 2c683b0..ebe0ce3 100644
--- a/app/src/ui/main_window.cpp
+++ b/app/src/ui/main_window.cpp
@@ -56,6 +56,9 @@ MainWindow::MainWindow()
     setGeometry(QApplication::primaryScreen()->availableGeometry());
 
     qInfo() << "Welcome to DesCartes Builder";
+
+    // FIXME
+    m_tabManager->openFrom("/home/eduardoconto/descartes-builder/app/examples/pipe-deformation.dcb");
 }
 
 MainWindow::~MainWindow()
